packages/mdc-textfield/foundation.js

```diff
@@ -83,6 +100,16 @@ export default class MDCTextfieldFoundation extends MDCFoundation {
     this.showHelptext_();
   }
 
+  autoCompleteFocus_(keyInteraction) {
+    if (!keyInteraction) {
+      this.activateFocus_();
+    }
+  }
+
+  setKeyInteraction_() {
+    this.keyInteractionOccurred = true;
```

Comment:
What happens in the following scenario:

Say there are two text fields, and the following events happen:

1. User begins typing into first textfield
1. Browser auto-completes for both the first and second textfield
1. User clears second textfield
1. User goes back to first textfield and begins typing
1. Browser auto-completes both first and second textfield _again_

In this case, since `keyInteractionOccured` is already set to `true` from the first go-around, would the autocomplete functionality work as expected?

I think we might have to use some more detailed heuristics in order to get this to work correctly. For example, is the element currently focused might be a better indication of whether or not the input event was triggered by an auto-complete rather than a user interaction (since un-focused elements can only be `input` by means of an auto-complete, I would imagine).